Uses a livereload plugin instead of relying on reloadPage

MiguelMadero opened the pull request:
The functionality is the same, but avoid the need to tweak `node_modules/ember-cli/node_modules/tiny-lr/node_modules/livereload-js/dist/livereload.js`. 

It also moves the inline JS to its own file to make it easier to maintain.

MiguelMadero commented:
Pused the fix for #1 here instead of opening a separate PR to avoid merge conflicts since I had to change similar areas.

MiguelMadero commented on README.md:
````diff
@@ -19,15 +19,6 @@ const Resolver = EmberResolver.extend(HotReloadMixin);
 export default Resolver;
 ```
 
-Add the development settings for ember-devtools to your config/environment.js
````
I only removed this and not the other reference to devtools ^^ to avoid conflicts with PR#2

toranb commented:
@MiguelMadero I like the idea of this very much (avoids any type of livereload node_modules hacking) but I'm unsure of "how" it actually works. Today the broccoli work fires off a WS* to the page/ that in turn triggers livereload-js to ask if it's css/js/etc and if JS we alert ember via devtools

Does this magical (as I'm still unsure how it works) plugin thing replace the broccoli lib work also then? If not how does `Plugin.prototype.reload = function(path, options) {` get fired from the broccoli WS* code ?

I'd like to wait on this until I understand it a little more if you don't mind :)

MiguelMadero commented:
This plugins in only client side, it doesn't deal with the broccoli side, ember-cli already starts the WS server and `lib/hot-reloader` does the rest as part of the CLI server middleware. 

I left a bit more on info on issue #2, see https://github.com/toranb/ember-cli-hot-loader/issues/3#issuecomment-242923180 , but, reloadPage is what we normally modified manually, livereload-js loops through all the plugin and calls reload before it calls `reloadPage`. It was already passing the path. If `reload` returns true it will stop the flow, otherwise continues to the next plugin and finally calls reloadPage if nothing else handled it. 

Here is the relevant block: https://github.com/MiguelMadero/livereload-js/blob/jshotreload/dist/livereload.js#L761-L786

toranb commented:
If that's the case this would be awesome - you've used this and we get the same 10ms all works all the time setup?

If so I could merge this and do a v0.0.2 release later tonight

Just wanted to see if you could rebase/ look over the readme one more time after my push (from just a few min ago)

MiguelMadero commented:
Sure, I did it in a way that it wouldn't conflict with the other README changes. It looks good, I rebased and had another look. 

> you've used this and we get the same 10ms all works all the time setup?

Yes. Same experience, I refactored it one step at a time, first extracting the file to get livereload and better dev experience, the switched to a plugin to avoid modifying node_modules, then finally to an initializer to remove the dependency on devTools and it kept working the same during  each step.

MiguelMadero commented:
Oops, I accidentally closed this.
